### PR TITLE
add connexion and login page merge dev in main 

### PR DESCRIPTION
**Pull request by pjroggy**

### CODE REVIEW

**Review by FGAT3 — APPROVED** [review]
Seems good to me

**Review by PaulBOURY — APPROVED** [review]

**Review by FGAT3 — APPROVED** [review]

**Review by JadeJolivet — APPROVED** [review]

File: client/src/components/FlipCard/FlipCard.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      transform: rotateY(0deg);
+    }
+  }
+
+  figcaption {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0.5rem;
+    box-sizing: border-box;
```
Comment (Yoann-Fortin) [inline client/src/components/FlipCard/FlipCard.scss]: repeated code

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          id="password"
+          name="password"
+          placeholder="Mot de passe"
+          pattern="/^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/"
+          required
+        />
+        <button
+          className="login-button"
+          type="submit"
+          aria-label="Se connecter"
```
Comment (Yoann-Fortin) [inline client/src/pages/LoginPage/LoginPage.jsx]: repeated code with content in this node

File: client/src/pages/RegistrationPage/RegistrationPage.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          id="password"
+          name="password"
+          placeholder="Mot de passe"
+          pattern="/^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/"
+          required
+        />
+        <button
+          className="validation-button"
+          type="submit"
+          aria-label="validation"
```
Comment (Yoann-Fortin) [inline client/src/pages/RegistrationPage/RegistrationPage.jsx]: same comment

File: server/app/middleware/authActions.js
Comment (Yoann-Fortin) [inline server/app/middleware/authActions.js:1]: This middleware has three reasons to change: it needs to be refactored into two middleswares and the controller needs to be left alone.

File: server/app/routers/api/router.js
Comment (Yoann-Fortin) [inline server/app/routers/api/router.js:1]: Missing another middleware and controller with specific method

File: server/database/models/UserRepository.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  // The C of CRUD - Create operation
+
+  async create(user) {
+    const avatarImage = this.avatar_image;
+    // Execute the SQL INSERT query to add a new user to the "user" table
+    const [result] = await this.database.query(
+      `INSERT INTO ${this.table} (
+        created_at,
```
Comment (Yoann-Fortin) [inline server/database/models/UserRepository.js]: dead code because your database already know a specific method to affect a correct date

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

**Review by Yoann-Fortin — APPROVED** [review]

**Pull request merged**